src/Formats/registerFormats.cpp

```diff
@@ -291,6 +291,8 @@ void registerFormats()
     registerDWARFSchemaReader(factory);
     registerOneSchemaReader(factory);
     registerNpySchemaReader(factory);
+
+    factory.registerAllFormatNames();
```

Comment:
This is wrong for 2 reasons:
* It's exactly backwards from the dozens of previous calls
* The formats are already registered. If we need another structure to do lookups, add it and change `registerInputFormat` / `registerRandomAccessInputFormat`/ whatever.

Reply:
I observed that `FormatFactory::file_extension_formats` can also be used as a case-insensitive format name mapping, so I implemented `FormatFactory::getOriginalFormatNameIfExists` to use it to look up farmat names. In addition, I created `KnownFormatNames::case_insensitive_names_map` to get the format name when parsing the FORMAT.